### PR TITLE
taplo: init

### PR DESCRIPTION
**Pull request by figsoda**
###### Motivation for this change

This pr adds taplo-cli and taplo-lsp

taplo is a TOML toolkit written in Rust

https://taplo.tamasfe.dev/
https://github.com/tamasfe/taplo

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -14396,6 +14396,10 @@ in
     stdenv = gccStdenv;
   };
 
+  taplo-cli = callPackage ../development/tools/taplo/cli.nix { };
+
+  taplo-lsp = callPackage ../development/tools/taplo/lsp.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:14401]:
```suggestion
  taplo-cli = callPackage ../development/tools/taplo-cli { };

  taplo-lsp = callPackage ../development/tools/taplo-lsp { };
```

File: pkgs/development/tools/taplo/lsp.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  cargoSha256 = "0ak70cwxcviv86b4zrcgqaxhdm6fxsji03mnacvp4pwlwv84ikkc";
+
+  # excludes test_tcp since it fails
+  cargoTestFlags = [ "test_stdio" ];
+
+  nativeBuildInputs = lib.optional stdenv.isLinux pkg-config;
+
+  buildInputs = lib.optional stdenv.isLinux openssl;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/taplo/lsp.nix:19]: Do you know if it build on darwin?

File: pkgs/development/tools/taplo/cli.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "0hh9l83z7qymakyf7ka756gwxpzirgdhf6kpzh89bcmpdfz70005";
+  };
+
+  cargoSha256 = "0bkpcnbrrfv07czs1gy8r9q1cp6fdfz2vmlfk9lsg3iapvyi5s1c";
+
+  nativeBuildInputs = lib.optional stdenv.isLinux pkg-config;
+
+  buildInputs = lib.optional stdenv.isLinux openssl;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/taplo/cli.nix:16]: Do you know if it builds on darwin?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please do one commit per package and follow the contributing guide when naming the commits.

**Comment by SuperSandro2000** [comment]
@ofborg build taplo-cli taplo-lsp

ofborg detects packages based on commit messages so naming is important.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131909` at 36c58761 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>taplo-cli</li>
    <li>taplo-lsp</li>
  </ul>
</details>

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please rename the files and after that we can merge.

File: pkgs/development/tools/taplo/cli.nix (hunk 3 above)
Comment (figsoda) [reply pkgs/development/tools/taplo/cli.nix:16]: It's not tested on darwin, but in one of the dependency only require openssl on linux
https://github.com/seanmonstar/reqwest#requirements

**Comment by figsoda** [comment]
sorry i didn't see that comment at first, now its fixed

File: pkgs/development/tools/taplo/cli.nix (hunk 3 above)
Comment (figsoda) [reply pkgs/development/tools/taplo/cli.nix]: if it doesn't build on darwin i'll make a patch

**Comment by figsoda** [comment]
@ofborg build taplo-cli taplo-lsp

File: pkgs/development/tools/taplo/cli.nix (hunk 3 above)
Comment (figsoda) [reply pkgs/development/tools/taplo/cli.nix]: darwin now builds after some changes

**Pull request merged**